client/v3/client_test.go

```diff
@@ -266,6 +268,108 @@ func TestSyncFiltersMembers(t *testing.T) {
 	}
 }
 
+func TestClientRejectOldCluster(t *testing.T) {
+	testutil.BeforeTest(t)
+	var tests = []struct {
+		name          string
+		endpoints     []string
+		versions      []string
+		expectedError error
+	}{
+		{
+			name:          "all new versions with the same value",
+			endpoints:     []string{"192.168.3.41:22379", "192.168.3.41:22479", "192.168.3.41:22579"},
+			versions:      []string{"3.5.4", "3.5.4", "3.5.4"},
+			expectedError: nil,
+		},
+		{
+			name:          "all new versions with different values",
+			endpoints:     []string{"192.168.3.41:22379", "192.168.3.41:22479", "192.168.3.41:22579"},
+			versions:      []string{"3.5.4", "3.5.4", "3.4.0"},
+			expectedError: nil,
+		},
+		{
+			name:          "all old versions with different values",
+			endpoints:     []string{"192.168.3.41:22379", "192.168.3.41:22479", "192.168.3.41:22579"},
+			versions:      []string{"3.3.0", "3.3.0", "3.4.0"},
+			expectedError: ErrOldCluster,
+		},
+		{
+			name:          "all old versions with the same value",
+			endpoints:     []string{"192.168.3.41:22379", "192.168.3.41:22479", "192.168.3.41:22579"},
+			versions:      []string{"3.3.0", "3.3.0", "3.3.0"},
+			expectedError: ErrOldCluster,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if len(tt.endpoints) != len(tt.versions) || len(tt.endpoints) == 0 {
+				t.Errorf(" endpoints & version length err")
```

Comment:
There is no any variadic parameters, so you shouldn't use `Errorf`. 

```suggestion
				t.Errorf("Unexpected endpoints and versions length, len(endpoints):%d, len(versions):%d", len(endpoints), len(versions))
```